src/js/components/campaignpage/CampaignPage.jsx

```diff
@@ -357,8 +357,20 @@ const CampaignPage = () => {
     console.log("ADS BEFORE CHARITY SORTING", ads);
 
     // initial donation record
-	const donation4charityUnscaled = yessy(campaign.dntn4charity)? campaign.dntn4charity : fetchDonationData({ ads });
-
+    let donation4charityUnscaled = yessy(campaign.dntn4charity)? campaign.dntn4charity : {};
+    const fetchedDonationData = fetchDonationData({ ads });
+    // Assign fetched data to fill holes
+    Object.keys(fetchedDonationData).forEach(cid => {
+        if (!donation4charityUnscaled[cid]) donation4charityUnscaled[cid] = fetchedDonationData[cid];
+    });
+    // Attach normalised IDs
+    Object.keys(donation4charityUnscaled).forEach(cid => {
+        const sogiveCid = normaliseSogiveId(cid);
+        if (!donation4charityUnscaled[sogiveCid]) {
+            donation4charityUnscaled[sogiveCid] = donation4charityUnscaled[cid];
```

Comment:
I think this could create "duplicate" entries in donation4charityUnscaled, which would then throw the scaling off.

Reply:
Added code to remove the old entries

Reply:
Let's normalise the ID in the previous loop e.g. line 364 instead.

Reply:
Thanks :)

Reply:
Done in latest commits